analytics/app/src/main/java/com/google/firebase/quickstart/analytics/kotlin/MainActivity.kt

```diff
@@ -73,22 +81,26 @@ class MainActivity : AppCompatActivity() {
         }
 
         // Create the adapter that will return a fragment for each image.
-        imagePagerAdapter = ImagePagerAdapter(supportFragmentManager, IMAGE_INFOS)
+        imagePagerAdapter = ImagePagerAdapter(supportFragmentManager, IMAGE_INFOS, lifecycle)
 
         // Set up the ViewPager with the pattern adapter.
         binding.viewPager.adapter = imagePagerAdapter
+        val viewPager: ViewPager2 = binding.viewPager
+        viewPager.adapter = imagePagerAdapter
 
-        // Workaround for AppCompat issue not showing ViewPager titles
-        val params = binding.pagerTabStrip.layoutParams as ViewPager.LayoutParams
-        params.isDecor = true
-
-        // When the visible image changes, send a screen view hit.
-        binding.viewPager.addOnPageChangeListener(object : ViewPager.SimpleOnPageChangeListener() {
+        val pageChangedCallback = object : ViewPager2.OnPageChangeCallback() {
             override fun onPageSelected(position: Int) {
                 recordImageView()
                 recordScreenView()
             }
-        })
+        }
+
+        binding.viewPager.registerOnPageChangeCallback(pageChangedCallback)
+
+        val tabLayout: TabLayout = binding.tabLayout
+        TabLayoutMediator(tabLayout, viewPager) { tab, position ->
```

Comment:
If the code repetition comment gets applied:
```suggestion
        TabLayoutMediator(tabLayout, binding.viewPager) { tab, position ->
```